### PR TITLE
404 page with layout based on app router

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Implement 404 page inside full page layout in app router.  Remove pages/404 since it's never used when we have anything defined in app router

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3953?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3929** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3936** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3936?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3939** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3939?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3893** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3893?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #3923** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3923?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #3948** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3948?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #3949** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3949?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #3953** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3953?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3953?utm_source=stack-comment).

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/app/[locale]/not-found.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { NotFoundPageContent } from './NotFoundPageContent'
+
+export default function NotFound() {
```
Comment (gustaveen) [inline apps/store/src/app/[locale]/not-found.tsx:3]: aha so `not-found` is the route for 404

**Review by gustaveen — APPROVED** [review]
Nice! 

I thing I thought about now while we will be migrating, we will end up with a bunch of app specific components inside the /app folder. Would be nice if we can keep a clear app route strucutre (makes it easier to get into that thinking) so maybe we can create a folder like `appComponents` to separate it?

**Pull request closed without merging**